core/controllers/admin_test.py

```diff
@@ -723,20 +722,21 @@ def test_update_flag_rules_with_rules_of_non_list_of_dict_type_returns_400(
         self.login(self.ADMIN_EMAIL, is_super_admin=True)
         csrf_token = self.get_new_csrf_token()
 
-        response = self.post_json(
-            '/adminhandler', {
-                'action': 'update_feature_flag_rules',
-                'feature_name': 'feature_name',
-                'new_rules': [1, 2],
-                'commit_message': 'test update feature',
-            },
-            csrf_token=csrf_token,
-            expected_status_int=400
-        )
         error_msg = (
             'Schema validation for \'new_rules\' failed: \'int\' '
             'object has no attribute \'__getitem__\'')
-        self.assertEqual(response['error'], error_msg)
+        with self.assertRaisesRegexp(Exception, error_msg):
```

Comment:
It's odd -- the changes in this file seem to be inconsistent (you're switching from method A to method B of checking an error, then switching from method B to method A in the next instance). Shouldn't we just standardize throughout on one way of doing this?

Reply:
Removed the `with` part.